pymatgen/command_line/mcsqs_caller.py

```diff
@@ -178,7 +178,11 @@ def _parse_sqs_path(path) -> Sqs:
     detected_instances = len(list(path.glob("bestsqs*[0-9]*.out")))
 
     # Convert best SQS structure to CIF file and pymatgen Structure
-    with Popen("str2cif < bestsqs.out > bestsqs.cif", shell=True, cwd=path) as process:  # noqa: S607
+    with (
+        open(os.path.join(path, "bestsqs.out")) as input_file,
+        open(os.path.join(path, "bestsqs.cif"), "w") as output_file,
+    ):
+        process = Popen(["str2cif"], stdin=input_file, stdout=output_file, cwd=path)  # noqa: S607
```

Comment:
Need to really carefully check this as well, as `mcsqs_caller` is not covered by unit test.